File: cs430/SPLProjectB_PetroulesJ/Source/SPLProjectB_PetroulesJ_CPP/doctorqueue.cpp
#include "doctorqueue.h"

DoctorQueue::DoctorQueue(const QString &name)
    : QQueue(), m_name(name)
{
}

QString DoctorQueue::name() const
{
    return this->m_name;
}

void DoctorQueue::setName(const QString &name)
{
    this->m_name = name;
}

bool DoctorQueue::isAvailable() const
{
    return this->m_isAvailable;
}

void DoctorQueue::setAvailable(bool available)
{
    this->m_isAvailable = available;
}

File: cs280/DSAss05A_PetroulesJ/src/dsass05a_petroulesj/DoctorQueue.java
package dsass05a_petroulesj;

import dslib_petroulesj.Queue;

/**
 *
 * @author Jake Petroules
 */
public class DoctorQueue extends Queue<ClinicPatient>
{
    private String name;

    private boolean available;
    
    public DoctorQueue(String name)
    {
        this.name = name;
    }

    public String getName()
    {
        return this.name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public boolean isAvailable()
    {
        return this.available;
    }

    public void setAvailable(boolean available)
    {
        this.available = available;
    }
}
